docs/developers.md

```diff
@@ -77,38 +77,59 @@ We suggest partially compiling DataHub according to your needs:
   ./gradlew :docs-website:serve
   ```
 
-## Deploying Local Versions
+##  Deploying Local Versions
+This guide explains how to set up and deploy DataHub locally for development purposes.
 
-Run just once to have the local `datahub` cli tool installed in your $PATH
+###  Initial Setup
+Before you begin, you'll need to install the local `datahub` CLI tool:
 
 ```shell
-cd smoke-test/
+cd metadata-ingestion/
 python3 -m venv venv
 source venv/bin/activate
-pip install --upgrade pip wheel setuptools
-pip install -r requirements.txt
 cd ../
 ```
 
-Once you have compiled & packaged the project or appropriate module you can deploy the entire system via docker-compose by running:
+### Deploying the Full Stack
 
+Deploy the entire system using docker-compose:
 ```shell
-./gradlew quickstart
+./gradlew quickstartDebug
 ```
 
-Replace whatever container you want in the existing deployment.
-I.e, replacing datahub's backend (GMS):
+Access the DataHub UI at `http://localhost:9002`
 
+### Refreshing the Frontend
+
+To run and update the frontend with local changes, open a new terminal and run:
 ```shell
-(cd docker && COMPOSE_DOCKER_CLI_BUILD=1 DOCKER_BUILDKIT=1 docker compose -p datahub -f docker-compose-without-neo4j.yml -f docker-compose-without-neo4j.override.yml -f docker-compose.dev.yml up -d --no-deps --force-recreate --build datahub-gms)
+cd datahub-web-react
+yarn install && yarn start
 ```
+The frontend will be available at `http://localhost:3000` and will automatically update as you make changes to the code.
 
-Running the local version of the frontend
+### Refreshing the GMS
 
+To refresh the GMS (Generalized Metadata Service) with local changes:
 ```shell
-(cd docker && COMPOSE_DOCKER_CLI_BUILD=1 DOCKER_BUILDKIT=1 docker compose -p datahub -f docker-compose-without-neo4j.yml -f docker-compose-without-neo4j.override.yml -f docker-compose.dev.yml up -d --no-deps --force-recreate --build datahub-frontend-react)
+./gradlew :metadata-service:war:build -x test --parallel && docker restart datahub-datahub-gms-debug-1
 ```
 
+### Refreshing Other Components
+
+To refresh other components with local changes:
```

Comment:
do we need a gradle command here?

Reply:
@hsheth2 hmm makes sense, i have a few questions tho

- Is refreshing other components common enough to justify adding gradle command?
- Is the current command the best way to refresh them or is there any better one? 

this section was meant to preserve the original doc -- if there’s a simpler approach, I’d love to include it there!

Reply:
mainly just that we need to call out the fact that you need to run a gradle command before restarting the docker compose container - just restarting isn't enough

Reply:
we don't need an exhaustive list of gradle commands. I just don't want people to get confused / only restart the container and then get confused

Reply:
Ahh i see,  TBH i don't have a full context here -- something like this would make sense? 

```shell
./gradlew quickstartDebug && (cd docker && COMPOSE_DOCKER_CLI_BUILD=1 DOCKER_BUILDKIT=1 docker compose -p datahub -f docker-compose-without-neo4j.yml -f docker-compose-without-neo4j.override.yml -f docker-compose.dev.yml up -d --no-deps --force-recreate --build <container_name>)
```

Reply:
maybe just say `# <run the correct gradle command to rebuild this component>`